modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java

```diff
@@ -991,6 +1014,11 @@ private Document _updateDocument(
 	private static final Log _log = LogFactoryUtil.getLog(
 		DocumentResourceImpl.class);
 
+	private static final Snapshot<SiteConnectedGroupGroupProvider>
+		_siteConnectedGroupGroupProviderSnapshot = new Snapshot<>(
+			DocumentResourceImpl.class, SiteConnectedGroupGroupProvider.class,
+			null, true);
+
```

Comment:
Why use a snapshot instead of injecting the reference?

Reply:
You're right, injecting the reference is a better way to do this. I was following the same pattern as DDMStructureLocalServiceImpl and DDMTemplateLocalServiceImpl, but the reference is better.